projects/nx-verdaccio/src/plugin/targets/create-targets.unit-test.ts

```diff
@@ -0,0 +1,249 @@
+import { beforeEach, describe, expect, type MockInstance } from 'vitest';
+import { type ProjectConfiguration } from '@nx/devkit';
+
+import * as nxDevkitMockModule from '@nx/devkit';
+
+import { createProjectConfiguration } from './create-targets';
+import {
+  TARGET_PACKAGE_INSTALL,
+  TARGET_PACKAGE_PUBLISH,
+} from './package.targets';
+import {
+  TARGET_ENVIRONMENT_E2E,
+  TARGET_ENVIRONMENT_SETUP,
+  TARGET_ENVIRONMENT_INSTALL,
+  TARGET_ENVIRONMENT_TEARDOWN,
+  TARGET_ENVIRONMENT_BOOTSTRAP,
+  TARGET_ENVIRONMENT_PUBLISH_ONLY,
+  TARGET_ENVIRONMENT_VERDACCIO_STOP,
+  TARGET_ENVIRONMENT_VERDACCIO_START,
+} from './environment.targets';
+
+import { type NxVerdaccioCreateNodeOptions } from '../schema';
+import { type NormalizedCreateNodeOptions } from '../normalize-create-nodes-options';
+
+import * as packageTargetsSpyModule from './package.targets';
+import * as environmentTargetsModule from './environment.targets';
+import * as normalizeCreateNodesSpyModule from './../normalize-create-nodes-options';
+
+describe('createProjectConfiguration', (): void => {
+  const implicitDependencies = ['mock-implicit-dep'];
+  const projectConfiguration: ProjectConfiguration = {
+    root: 'mock-root',
+    name: 'unit-test-project',
+    targets: { build: {} },
+  };
+  const options: NxVerdaccioCreateNodeOptions = {
+    environments: {
+      targetNames: ['build'],
+    },
+  };
+  const normalizedOptions: NormalizedCreateNodeOptions = {
+    environments: {
+      targetNames: ['build'],
+      environmentsDir: './environments',
+    },
+    packages: {
+      targetNames: ['test', 'lint'],
+      environmentsDir: './packages',
+    },
+  };
+
+  vi.mock('@nx/devkit', () => {
+    return {
+      logger: {
+        warn: vi.fn(),
+      },
+    };
+  });
+
+  let normalizeCreateNodesOptionsSpy: MockInstance;
+  let isEnvProjectSpy: MockInstance;
+  let isPkgSpy: MockInstance;
+  let verdaccioTargetsSpy: MockInstance;
+  let getEnvTargetsSpy: MockInstance;
+  let updateEnvTargetNamesSpy: MockInstance;
+
+  beforeEach((): void => {
+    normalizeCreateNodesOptionsSpy = vi
+      .spyOn(normalizeCreateNodesSpyModule, 'normalizeCreateNodesOptions')
+      .mockReturnValue(normalizedOptions);
+    isEnvProjectSpy = vi
+      .spyOn(environmentTargetsModule, 'isEnvProject')
+      .mockReturnValue(true);
+    isPkgSpy = vi
+      .spyOn(packageTargetsSpyModule, 'isPkgProject')
+      .mockReturnValue(true);
```

Comment:
General best practices for setup spy's and configuring spy's:

- setup in describe
- configuring with potential defailts in beforeEach
- actual mock return in the it block